mzLib/Transcriptomics/NucleicAcid.cs

```diff
@@ -40,39 +41,58 @@ public abstract class NucleicAcid : INucleicAcid, IBioPolymer, IEquatable<Nuclei
         #region Constuctors
 
         /// <summary>
-        /// For creating an RNA programatically
+        /// For creating an RNA programatically. Filters out modifications that do not match their nucleotide target site.
         /// </summary>
-        protected NucleicAcid(string sequence, IHasChemicalFormula? fivePrimeTerm = null, IHasChemicalFormula? threePrimeTerm = null,
-            IDictionary<int, List<Modification>>? oneBasedPossibleLocalizedModifications = null)
+        protected NucleicAcid(string sequence,
+            IDictionary<int, List<Modification>>? oneBasedPossibleLocalizedModifications = null,
+            IHasChemicalFormula? fivePrimeTerm = null, IHasChemicalFormula? threePrimeTerm = null)
         {
+            NonVariant = this;
             MonoisotopicMass = 0;
             _nucleicAcids = new Nucleotide[sequence.Length];
-            ThreePrimeTerminus = threePrimeTerm ??= DefaultThreePrimeTerminus;
-            FivePrimeTerminus = fivePrimeTerm ??= DefaultFivePrimeTerminus;
-            _oneBasedPossibleLocalizedModifications = oneBasedPossibleLocalizedModifications ?? new Dictionary<int, List<Modification>>();
-            GeneNames = new List<Tuple<string, string>>();
-
+            ThreePrimeTerminus = threePrimeTerm ?? DefaultThreePrimeTerminus;
+            FivePrimeTerminus = fivePrimeTerm ?? DefaultFivePrimeTerminus;
             ParseSequenceString(sequence);
+
+            SequenceVariations = [];
+            AppliedSequenceVariations = [];
+            TruncationProducts = [];
+            OriginalNonVariantModifications = oneBasedPossibleLocalizedModifications ?? new Dictionary<int, List<Modification>>();
+            OneBasedPossibleLocalizedModifications = oneBasedPossibleLocalizedModifications != null 
+                ? ((IBioPolymer)this).SelectValidOneBaseMods(oneBasedPossibleLocalizedModifications) 
+                : new Dictionary<int, List<Modification>>();
         }
 
         /// <summary>
-        /// For Reading in from rna database
+        /// For Reading in from rna database. Filters out modifications that do not match their nucleotide target site.
         /// </summary>
-        protected NucleicAcid(string sequence, string name, string identifier, string organism, string databaseFilePath,
-            IHasChemicalFormula? fivePrimeTerm = null, IHasChemicalFormula? threePrimeTerm = null,
+        protected NucleicAcid(string sequence, string accession,
```

Comment:
does the generic "NucleicAcid" apply equivalently to deoxyribonucleic acid (DNA) and ribonucleic acid (RNA)?

Reply:
Yes

Reply:
yes